packages/api/src/controllers/stripe.ts

```diff
@@ -214,6 +225,70 @@ const sendUsageRecordToStripe = async (
   );
 };
 
+export const calculateOverageOnMinimumSpend = async (product, billingUsage) => {
+  const minimumSpend = product.monthlyPrice;
+
+  let storageUsage = billingUsage.storageUsage;
+  let deliveryUsage = billingUsage.deliveryUsage;
+  let totalUsage = billingUsage.totalUsage;
+
+  let prices: any = {};
+
+  if (product?.usage) {
+    product.usage.forEach((item) => {
+      if (item.name.toLowerCase() === "transcoding")
+        prices.transcoding = item.price;
+      if (item.name.toLowerCase() === "delivery") prices.streaming = item.price;
+      if (item.name.toLowerCase() === "storage") prices.storage = item.price;
+    });
+  }
+
+  let totalSpent =
+    prices.storage * storageUsage +
+    prices.streaming * deliveryUsage +
+    prices.transcoding * totalUsage;
+  if (totalSpent > minimumSpend) {
+    let remainingOverage = totalSpent - minimumSpend;
+    let overageUsage = {
+      StorageUsageMins: 0,
+      DeliveryUsageMins: 0,
+      TotalUsageMins: 0,
+    };
+
+    if (remainingOverage > 0) {
+      const totalSpentInCategories = {
+        storage: prices.storage * storageUsage,
+        delivery: prices.streaming * deliveryUsage,
+        transcoding: prices.transcoding * totalUsage,
+      };
+
+      const totalSpent =
+        totalSpentInCategories.storage +
+        totalSpentInCategories.delivery +
+        totalSpentInCategories.transcoding;
+
+      if (totalSpent > 0) {
+        const overageRatio = remainingOverage / totalSpent;
```

Comment:
This seems like some pretty complex logic. Is it described anywhere? The description you gave on the PR doesn't seem to describe this stuff like "overage ratio". Maybe worth a comment in this function declaration explaining how it works wdyt?

Reply:
I added several comments across the steps!
This is the part where I mostly need the feedback, so I add some description here:

Stripe does not have a built-in way to implement minimum spend for metered usage, so if I report 30$ of usage to it on a product with a 100$ fixed monthly price, it will produce a 130$ invoice. Before these changes, we had fixed amount of minutes included in the plan, so whenever one of the items (transcoding, storage or delivery) had an overage, we would have reported it.

Instead, now, we don't have a fixed amount of minutes, we do have instead a fixed amount of dollars included in the plan, with a minimum monthly spend. So, if you use 1$ or 99$ or anything under 100$ across all three product items, you will receive an invoice of 100$.

This has an implication: 
Whenever the 100$ is exceeded, we need to calculate the amount of minutes in all three product items to subtract the included amount. To do this (with this proposed change) we take the total overage (i.e. 3$ out of a 103$ total usage) and distribute it across the items that have more than 0$ usage.

Stripe is going to get reported usage rows for 3$ in this example